platforms/core-runtime/native/src/main/java/org/gradle/platform/internal/DefaultBuildPlatform.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.platform.internal;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import net.rubygrapefruit.platform.SystemInfo;
+import org.gradle.api.GradleException;
+import org.gradle.platform.Architecture;
+import org.gradle.platform.BuildPlatform;
+import org.gradle.platform.OperatingSystem;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Objects;
+
+public class DefaultBuildPlatform implements BuildPlatform, Serializable {
```

Comment:
❓ How would users create one of those?

Reply:
Damn, codes lives in a Java 6 entry point, can't add a factory method on the interface. Added a separate factory for now. Will refactor after Gradle 9.